sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -560,7 +560,6 @@ public Instant timestamp(DoFn<InputT, OutputT> doFn) {
         throw new IllegalStateException(
             String.format("Unknown URN %s", pTransform.getSpec().getUrn()));
     }
-    this.onTimerContext = new OnTimerContext();
```

Comment:
The intent of these context objects was to not create a new one when processing each element/timer and instead to reference a member variable as can be seen in:
https://github.com/apache/beam/blob/591de3473144de54beef0932131025e2a4d8504b/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java#L1468
https://github.com/apache/beam/blob/591de3473144de54beef0932131025e2a4d8504b/sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java#L1506

It would have made more sense to have OnTimerContext just return `currentTimer.getUserKey()`

Reply:
+1 Let's fix these in a follow-up.

Reply:
And document the intent.

Reply:
> And document the intent.

! :)

@rehmanmuradali Do you have time to look into this?

Reply:
+1 worth to create a JIRA to track it.

Reply:
@mxm  I am currently working on BEAM-1589. I will get back to it after that. The Jira is filed here: https://issues.apache.org/jira/browse/BEAM-9839